ci: remove material-unit-tests failure blocklist

devversion opened the pull request:
* See individual commits

devversion committed:
ci: remove material-unit-tests failure blocklist

Initially when the `material-unit-tests` job got wired up,
Ivy was not really backwards-compatible and a few bugs caused
test failures when running the Angular Material test suites w/ Ivy.

These bugs got fixed progressively and eventually the test
blocklist became empty. At this point we don't want to regress
in the future and the blocklist should never have new items.

Additionally since we switched the unit-tests job to run against
Angular Material `master` with Bazel, the blocklist is no
longer respected. Therefore we can safely remove the blocklist.

JoostK commented on docs/DEBUG_MATERIAL_IVY.md:
````diff
@@ -0,0 +1,35 @@
+**important** to specify the `--define=compile=aot` flag in order to run tests with Ivy.
+  
+```bash
+yarn bazel --define=compile=aot src/material/slider:unit_tests
````
Shouldn't these be

```suggestion
yarn bazel test --define=compile=aot src/material/slider:unit_tests
```

devversion replied:
Ops yeah. Fixed :smile:

gkalpak reviewed (approved):
🚀

kara commented on docs/DEBUG_MATERIAL_IVY.md:
```diff
@@ -0,0 +1,35 @@
+# Debugging the Material unit tests job
+
+Currently all changes to Ivy are validated against the test suite of the Angular
+components repository. In order to debug the `material-unit-tests` CI job, the
```
Nit: might help to be more specific about where it lives

```suggestion
@ angular/components repository. In order to debug the `material-unit-tests` CI job, the
```

kara commented on docs/DEBUG_MATERIAL_IVY.md:
```diff
@@ -0,0 +1,35 @@
+following steps can be used:
+
+1\) Build the Ivy package output by running `./scripts/build-ivy-npm-packages.sh`.
+2\) Set up the package output in the `components` repository by running:
```
When I first read this, I thought you meant to run it in the components repository. Can we add:

```suggestion
2\) Set up the package output in the `components` repository by running (in this repo):
```

kara replied:
Can we also add a step before this to clone down the repo if you don't already have it?

kara commented on docs/DEBUG_MATERIAL_IVY.md:
```diff
@@ -0,0 +1,35 @@
+selecting a given test target.
+
+Here is an example of commands that run individual test targets. Note that it is
+**important** to specify the `--define=compile=aot` flag in order to run tests with Ivy.
```
👍 This is super helpful

kara reviewed (approved):
LGTM, a few things

devversion committed:
docs: add guide to reproduce material-unit-test failures locally

Adds a new guide that can be used to reproduce failures
reported in the `material-unit-tests` job locally.

The document should live in the framework repository as
the package building scripts are local to the framework
repository.

kara reviewed (approved):
LGTM

kara commented:
merge-assistance: global approval

devversion commented:
Closing this in favor of a new PR.
